### DISCUSSION TITLE
generate tf file based on a blueprint subaccount

### DISCUSSION THREAD

**Discussion by shaoyongfeng** [SAP/terraform-provider-btp#389] [Q&A] [answered]
is it possible to generate tf file based on a blueprint subaccount and then use it to create a new subacount?

**Comment by lechnerc77** [comment, marked as answer]
Not fully automated, but with Terraform version 1.5 the so called *import blocks* have been introduced. With this you can import existing accounts to Terraform this is a very handy evolution of the old `terraform import` statement. 
In order to get your resource definition you must also use the (currently) experimental `-generate-config-out` flag. 

Based on that import that is intended to bring existing infrastructure under management by Terraform you  can refine your setup manually (like refactoring into Terraform modules etc.) to have your blueprint for Terraform available. However, some manual work needed to do so. 

Depending on the complexity of your blueprint account it might be easier to create the files manually, instead of using the aforementioned import functionality.

Here is the link to the documentation of the import blocks: <https://developer.hashicorp.com/terraform/language/import>
Here is a short video to see the import blocks as well as the experimental `-generate-config-out` flag in action: <https://youtu.be/znfh_00EDZ0?si=UJdKgxWuAI5iwHmv>
